src/components/Carousel.vue

```diff
@@ -0,0 +1,276 @@
+        const stepSize = 2 * Math.PI / this._totalPositionCount;
+        const entryIndex = this.entries.indexOf(entry);
+        const selectedIndex = this.entries.indexOf(this.effectiveSelected);
+        let offset = entryIndex - selectedIndex;
+        if (this._hasDummyPosition && offset > this._totalPositionCount / 2) {
+            // skip dummy position
+            offset += 1;
+        }
+        return currentRotation + this._calculateRotationInClosestDirection(currentRotation, offset * stepSize);
+    }
+
+    @Watch('hideBackgroundEntries')
+    private _rerender() {
+        if (this.requestAnimationFrameId !== null) return;
+        this.requestAnimationFrameId = requestAnimationFrame(() => {
+            const zCoordinatesForEntries: Array<[string, number]> = [];
+            let finished = this.radius.finished;
+            for (const [entry, rotation] of this.rotations) {
+                const currentRotation = rotation.currentValue;
+                const currentRadius = this.radius.currentValue;
+                const x = Math.sin(currentRotation) * currentRadius;
+                const z = Math.cos(currentRotation) * currentRadius - currentRadius;
+                const [ el ] = this.$refs[entry];
+                el.style.transform = `translate3d(calc(${x}px - 50%),-50%,${z}px)`;
+                el.style.display = this._shouldHide(entry) ? 'none' : '';
+                zCoordinatesForEntries.push([entry, z]);
+                finished = finished && rotation.finished;
+            }
+
+            zCoordinatesForEntries.sort(([, z1], [, z2]) => z1 - z2);
+            for (let i = 0; i < zCoordinatesForEntries.length; ++i) {
+                const [ el ] = this.$refs[zCoordinatesForEntries[i][0]];
+                el.style.zIndex = `${i}`;
+            }
+
+            this.requestAnimationFrameId = null;
+            if (!finished) this._rerender();
+        });
+    }
+
+    private _calculateRotationInClosestDirection(fromAngle, toAngle): number {
+        // angle offset modulo full rotations
+        const rotation = (toAngle - fromAngle) % (2 * Math.PI);
+        // determine rotation in opposite direction (subtracting or adding a full circle depending on direction (sign))
+        const rotationOppositeDirection = rotation - Math.sign(rotation) * 2 * Math.PI;
+        if (Math.abs(Math.abs(rotation) - Math.abs(rotationOppositeDirection)) < 1e-10) {
+            // in case of ambiguity chose a default direction
+            return Math.min(rotation, rotationOppositeDirection);
+        } else if (Math.abs(rotation) < Math.abs(rotationOppositeDirection)) {
+            return rotation;
+        } else {
+            return rotationOppositeDirection;
+        }
+    }
+
+    private _shouldHide(entry): boolean {
+        const rotation = this.rotations.get(entry);
+        if (!rotation) return false;
+        const absoluteRotation = Math.abs(this._calculateRotationInClosestDirection(0, rotation.currentValue));
```

Comment:
This method should `return false` early `if(!this.disabled && !this.hideBackgroundEntries)` to avoid unnecessarily calculcating `absoluteRotation` when not used.